maintenance/wikia/sync_user_groups.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+use Wikia\DependencyInjection\Injector;
+use Wikia\UserGroups\Maintenance\UserGroupUpdater;
+
+$wgCommandLineSilentMode = true;
+
+require_once(dirname(__FILE__).'/../Maintenance.php');
+
+class SyncWikiUserGroups extends Maintenance {
+
+	public function __construct() {
+		parent::__construct();
+
+		$this->addOption('wiki_id', '', true, true);
+	}
+
+	public function loadParamsAndArgs($self = null, $opts = null, $args = null) {
+		parent::loadParamsAndArgs($self, $opts, $args);
+
+		if (!empty($this->mOptions['wiki_id'])) {
+			putenv("SERVER_ID={$this->mOptions['wiki_id']}");
+		}
+	}
+
+	public function execute() {
+		global $wgCityId;
+
+		/** @var UserGroupUpdater $updater */
+		$updater = Injector::getInjector()->get(UserGroupUpdater::class);
+		$updater->updateForWiki($wgCityId, wfGetDB(DB_SLAVE));
```

Comment:
I'd rather use the `Injector` here:
* Creating a `UserGroupUpdater` instance would effectively require duplicating all the logic in `UserGroupsModule`.
* There is overhead in _creating_ the `Injector`, sure, but that overhead has already been paid at this point (it happens during the MW bootstrap phase), so there's no real additional overhead by _using_ the `Injector` here.
* With regards to testing, I'd rather have any complexities in setting up the `Injector` state properly in the tests rather than in the source. The `Injector` is created at the end of `LocalSettings.php`, so if `WikiaBaseTest` goes through that code (I think it does - I'm not sure), the injector will be properly initialized. If not, or if the test wants to initialize the `Injector` with a mock, the test can explicitly do that with `Injector::setInjector` to make sure its dependencies are properly set. This is better than relying on overwriting `$wg*` variables because `Injector::setInjector` only has the dependencies explicitly loaded into it at call-time, whereas the `$wg*` variables carry over from test to test, so there's a chance there for tests to cause each other to fail depending on the order in which they are run (which we have definitely had trouble with in the past). See [UserTest](https://github.com/Wikia/app/blob/dev/includes/wikia/tests/UserTest.php#L43) for an example of the injector being used in tests.